src/meili/traffic_segment_matcher.cc

```diff
@@ -153,7 +154,14 @@ std::list<std::vector<interpolation_t> > TrafficSegmentMatcher::interpolate_matc
       //get the distance and match result for the begin node of the edge
       interpolated.emplace_back(interpolation_t{segment->edgeid, total_length, 0.f, last_index, -1});
       //add distances for all the match points that happened on this edge
-      for(; i < matches.size() && matches[i].edgeid == segment->edgeid; ++i) {
+      for(; i < matches.size(); ++i) {
+        //skip unroutable ones, we dont know what edge they were on
+        if(!matches[i].edgeid.Is_Valid())
```

Comment:
turns out you can have match points that end up with in valid state ids in the middle of a trace that does give you back a full route. because i didnt realize that a route could connect across invalid states i wasnt even checking if they were valid! this lead to not being able to interpolate time values starting where the invalid state was and continuing for the rest of the trace. what this meant was that we'd lose out on reporting any ots's that came after that invalid state. this fix will increase the number of segments we report on